### PR TITLE
Adds support to adoCodespacesAuth.tenantID extension setting for conf…

### PR DESCRIPTION
**Pull request by liguori**
Provides the 'adoCodespacesAuth.tenantID' extension setting to configure the Entra ID tenant to sign in to.

Address #14

### CODE REVIEW

File: src/extension.ts [hunk 1]
```diff
@@ -110,7 +110,14 @@ const createHelperExecutable = (
 
 const authenticateAdo = async (context: vscode.ExtensionContext) => {
   try {
-    await getAccessToken();
+    var scopes = ["499b84ac-1321-427f-aa17-267ca6975798/.default"];
+
+    var tenantID = vscode.workspace.getConfiguration("adoCodespacesAuth").get('tenantID');
```
Comment (ronakj) [inline src/extension.ts:115]: `var` is now considered obsolete in most modern JS codesbases. Prefer `let` or `const`.

File: src/extension.ts [hunk 2]
```diff
@@ -110,7 +110,14 @@ const createHelperExecutable = (
 
 const authenticateAdo = async (context: vscode.ExtensionContext) => {
   try {
-    await getAccessToken();
+    var scopes = ["499b84ac-1321-427f-aa17-267ca6975798/.default"];
```
Comment (ronakj) [inline src/extension.ts:113]: I would love it if we could move this to a Constant variable declared somewhere at the top of this file. Something like `const DEFAULT_ADO_SCOPE = "499b84ac-1321-427f-aa17-267ca6975798/.default";`

**Review by ronakj — CHANGES_REQUESTED** [review]

**Comment by ronakj** [comment]
Mostly looks good functionally. Just some minor code changes and we should be good to go. 

Can you also update the `package.json` version (probably a minor bump to 1.2.0) and add a line in the `README.md` to explain this configuration? Thanks!

File: src/extension.ts (hunk 2 above)
Comment (liguori) [reply src/extension.ts]: Done in new commit

**Review by ronakj — APPROVED** [review]

**Comment by ronakj** [comment]
Thanks @liguori. I will do a quick test from my side and then release it.

**Comment by ronakj** [comment]
When testing this, it seems to not work when requesting token. It authenticates fine initially, but when you request a token with the command `ado-auth-helper get-access-token` or with `git fetch` it just hangs. 

There are two reasons from what I can see. The PR breaks the default assumption that `getAccessToken` when called without scopes uses the ADO scope. So when called from socket server handler it doesn't get the default scope now. Another reason might be because we need to pass the tenant in the calls to `getAccessToken` function when it's called from socket server handler. 

@liguori can you create a followup PR. Maybe move the logic for getting tenant from configuration to the `getAccessToken` function and also fix the default scope assumption.

You can test it by updating the version number, running `yarn vsce:package`, download the generated vsix bundle and then open a codespace, search for "vsix" and install the local download

image

Then signout, add the tenant id config and then signin to microsoft account. Run `ado-auth-helper get-access-token` to test if it works.

**Comment by ronakj** [comment]
It's late for me today so I am gonna go, but I can also take a stab at fixing this tomorrow if you want.

**Pull request merged**